src/api/tasks.coffee

```diff
@@ -128,12 +158,71 @@ exports.getTask = (taskId) ->
   taskId = unescape taskId
 
   try
-    # Try to get the Task (Call the function that emits a promise and Koa will wait for the function to complete)
-    result = yield Task.findById(taskId).exec()
+
+    filtersObject = this.request.query
+
+    #get limit and page values
+    filterLimit = filtersObject.filterLimit
+    filterPage = filtersObject.filterPage
+
+    #determine skip amount
+    filterSkip = filterPage*filterLimit
+    
+    # get filters object
+    filters = JSON.parse filtersObject.filters
+
+    result = yield Task.findById(taskId).lean().exec()
+    tempTransactions = result.transactions
+
+    # are filters present
+    if Object.keys( filters ).length > 0
```

Comment:
This method's getting a little big and it might be a good idea to split it up into smaller units of work. This section looks like a good candidate. This could easily be split into a method `buildFilteredTransactionsArray` or something similar.